src/main/java/com/synopsys/integration/alert/web/security/authentication/AlertAuthenticationProvider.java

```diff
@@ -47,12 +50,19 @@ public Authentication authenticate(Authentication authentication) throws Authent
             throw new IllegalArgumentException("Only UsernamePasswordAuthenticationToken is supported, " + authentication.getClass() + " was attempted");
         }
 
-        return authenticationPerformers
-                   .stream()
-                   .map(authPerformer -> authPerformer.performAuthentication(authentication))
-                   .flatMap(Optional::stream)
-                   .findAny()
-                   .orElse(authentication);
+        for (AuthenticationPerformer authenticationPerformer : authenticationPerformers) {
+            try {
+                Optional<Authentication> completedAuthentication = authenticationPerformer.performAuthentication(authentication);
+                if (completedAuthentication.isPresent() && completedAuthentication.get().isAuthenticated()) {
```

Comment:
You could add `.filter(Authentication::isAuthenticated)` to avoid chaining methods after a `.get()`.